### PR TITLE
Avoid sending "BlobOnlyStorageWrongCall" event while getting policies

### PR DESCRIPTION
**Pull request by dannimad**
While creating a detached container, containerRuntime constructor tries to get storage policies for optional variable maxSnapshotCacheDurationMs. In case of having a BlobOnlyStorage, this will always trigger error "BlobOnlyStorage not implemented method used" which is noisy, so instead of calling notCalled function while getting policies, just return undefined. Added more information to the event in case there are other reasons why the event is being triggered this often.

### CODE REVIEW

**Comment by anthony-murphy** [comment]
i'm not sure this is the right fix. will the detached runtime get the right information once it attaches? i wonder if a better fix would be to have the runtime not try to access this until it is attached

**Comment by dannimad** [comment]
> i'm not sure this is the right fix. will the detached runtime get the right information once it attaches? i wonder if a better fix would be to have the runtime not try to access this until it is attached

AFAIK once the container is attached, we will rely on real document storage instead of BlobOnlyStorage, so we should be able to get the right information since I'm only changing the latter. The policies request happens on the constructor so there is no waiting here to be done, we either have the policies available or we get undefined.

**Comment by anthony-murphy** [comment]
it is wrong to try and access the policies in the constructor of the container runtime, as they may not be available, whatever they are used for need to wait util the container is determined to be attached before it can access them

**Review by anthony-murphy — APPROVED** [review]

File: packages/loader/container-loader/src/containerStorageAdapter.ts [hunk 1]
```diff
@@ -190,28 +190,33 @@ class BlobOnlyStorage implements IDocumentStorageService {
...
+		return this.notCalled("repositoryUrl");
 	}
 
 	/* eslint-disable @typescript-eslint/unbound-method */
-	public getSnapshotTree: () => Promise<ISnapshotTree | null> = this.notCalled;
-	public getVersions: () => Promise<IVersion[]> = this.notCalled;
-	public write: () => Promise<IVersion> = this.notCalled;
-	public uploadSummaryWithContext: () => Promise<string> = this.notCalled;
-	public downloadSummary: () => Promise<ISummaryTree> = this.notCalled;
+	public getSnapshotTree: () => Promise<ISnapshotTree | null> = () =>
```
Comment (andre4i) [inline packages/loader/container-loader/src/containerStorageAdapter.ts:201]: I understand the convenience of having this in the event as a property, but in the interest of keeping the bundle size low and maybe the telemetry database size, don't we already have this information in the stack trace?

Something like:
```
union Office_Fluid_FluidRuntime_*
| where Event_Time > ago(1d)
| where Data_eventName == "fluid:telemetry:Container:BlobOnlyStorageWrongCall"
| summarize count() by Data_stack
| order by count_ desc 
| limit 10
```
will likely get you the most relevant scenarios to investigate.

**Review by andre4i — APPROVED** [review]

**Pull request merged**